### PR TITLE
JohnPapa Style Guide : Account files

### PR DESCRIPTION
**Pull request by wmarques**
Refactored all account files (activate, password, register, reset, sessions, settings, social) following JohnPapa's rules.

For a reason that I ignore, I couldn't wrap the _account.state.js into a closure function, this was causing routing bug : "cannot resolve 'route' state from 'app'" when clicking on an account link like Settings or Register. 

I also updated tests and they are fully working on the apps I've generated.

### CODE REVIEW

File: generators/client/templates/src/main/webapp/app/account/password/_password-strength-bar.directive.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+                template: '<div id="strength">' +
+                    '<small<% if (enableTranslation) { %> translate="global.messages.validate.newpassword.strength"<% } %>>Password strength:</small>' +
+                    '<ul id="strengthBar">' +
+                    '<li class="point"></li><li class="point"></li><li class="point"></li><li class="point"></li><li class="point"></li>' +
+                    '</ul>' +
+                    '</div>',
+                link: linkFunc
+            };
+
+            return directive;
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/account/password/_password-strength-bar.directive.js]: shouldnt the return be called after the below function?

Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/account/password/_password-strength-bar.directive.js]: ill change this

Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/account/password/_password-strength-bar.directive.js]: ok this is as per guideline

**Comment by deepu105** [comment]
btw @wmarques this should be the correct style for route files, ill update them for account and entity

```
(function() {
    'use strict';

    angular
        .module('<%=angularAppName%>')
        .config(stateConfig);

    function stateConfig($stateProvider) {
        $stateProvider
        .state('account', {
            abstract: true,
            parent: 'app'
        });
    }
})();
```

**Comment by deepu105** [comment]
sorry it should be this, i missed the inject

```
(function() {
    'use strict';

    angular
        .module('<%=angularAppName%>')
        .config(stateConfig);

    stateConfig.$inject = ['$stateProvider'];

    function stateConfig($stateProvider) {
        $stateProvider
        .state('account', {
            abstract: true,
            parent: 'app'
        });
    }
})();
```

**Comment by deepu105** [comment]
I have updated this for the account module

**Comment by deepu105** [comment]
and for entity state

**Comment by wmarques** [comment]
Thanks I'm going to follow your example for the other states !

**Pull request merged**